controllers/menuController.js

```diff
@@ -0,0 +1,95 @@
+const Menu = require("./../models/Menu");
+const jwt = require("jsonwebtoken");
+
+
+
+// Get all menu item
+
+const getAllMenu = async (req, res) => {
+    try {
+
+        const merchantId = req.merchantId;
+
+        const allMenuItems = await Menu.find({ merchantId });
+
+        if (!allMenuItems?.length) {
+            return res.status(400).json({ message: "No menu available" });
+        }
+        res.status(200).json({ success: true, data: allMenuItems });
+
+    } catch (error) {
+        console.error('Error getting all menu items:', error);
+        res.status(500).json({ success: false, error: 'Internal server error' });
+    }
+};
+
+
+
+// add new menu item
+
+const addMenuItem = async (req, res) => {
+    try {
+        const {
+            imageUrl,
+            name,
+            originalPrice,
+            description,
+            cuisineTypeId,
+            isFeatured
+        } = req.body;
+
+        const merchantId = req.merchantId;
+
+        const newMenuItem = new Menu({
+            imageUrl,
+            name,
+            originalPrice,
+            description,
+            merchantId,
+            cuisineTypeId,
+            isFeatured,
+        });
+
+        const savedMenuItem = await newMenuItem.save();
+
+        res.status(201).json({ success: true, data: savedMenuItem });
```

Comment:
remove success:true, since status 201 means successfully created. Also, insteaad of .save(), use .create({}); check other codes for the implementation.